### PR TITLE
VM Ownership code unification for Infra/cloud

### PR DESCRIPTION
**Pull request by rananda**
VM Ownership code unification for Infra/cloud using testgen

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit 3aedfef16c0edd4da2dc248568ed645210d75e64:

`cfme/tests/cloud_infra_common/test_vm_ownership.py`:
- :bangbang: Line 12:13: **E126** *continuation line over-indented for hanging indent*
- :bangbang: Line 142:101: **E501** *line too long (103 > 100 characters)*

Please, rectify these issues :smirk: .
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 6590d4c71936b4b0447b6d5c7d79d7fb2905b2f7:

Everything seems all right :smile: .
*CFME QE Bot*

File: cfme/tests/cloud_infra_common/test_vm_ownership.py [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+from utils import testgen
+
+
+def pytest_generate_tests(metafunc):
+    argnames, argvalues, idlist = testgen.all_providers(
+        metafunc, 'ownership_vm', require_fields=True)
+    testgen.parametrize(metafunc, argnames, argvalues, ids=idlist, scope="module")
+
+
+@pytest.fixture(scope="module")
```
Comment (psav) [inline cfme/tests/cloud_infra_common/test_vm_ownership.py]: This shouldn't be required, as the "setup_provider" should be used alongside testgen.

File: cfme/tests/cloud_infra_common/test_vm_ownership.py [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+    login.login_admin()
+    user2 = new_user(group_only_user_owned)
+    yield user2
+    login.login_admin()
+    user2.delete()
+
+
+@pytest.yield_fixture(scope="module")
+def user3(group_user_or_group_owned):
+    login.login_admin()
```
Comment (psav) [inline cfme/tests/cloud_infra_common/test_vm_ownership.py]: These lines are not needed as the create function has a login as admin

File: cfme/tests/cloud_infra_common/test_vm_ownership.py [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+                   email='abc@redhat.com',
+                   group=group_only_user_owned,
+                   cost_center='Workload',
+                   value_assign='Database')
+    user.create()
+    return user
+
+
+def test_form_button_validation(request, user1, setup_provider, provider, ownership_vm):
+    set_ownership_to_user = VM.factory(ownership_vm, provider)
```
Comment (psav) [inline cfme/tests/cloud_infra_common/test_vm_ownership.py]: I don't like these names, they imply a function, it's a just a vm object.

**Comment by dajoRH** [comment]
Lint report for commit f48984788e9064f9f33886caaabeb42839e209c9:

`cfme/tests/cloud_infra_common/test_vm_ownership.py`:
- :bangbang: Line 78:1: **E303** *too many blank lines (3)*

Please, rectify these issues :smirk: .
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 75019be7ae96ef7792612cc36b839bb2fe80485a:

Everything seems all right :smile: .
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit c020ac4a933303d139b13dfb4a9e76a0c5a3c23e:

`cfme/tests/cloud_infra_common/test_vm_ownership.py`:
- :red_circle: Line 137:21: **F821** *undefined name 'set_ownership_to_group'*

Please, rectify these issues :smirk: .
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit aa54160ecb24272c44e947249c0dae688a6ed591:

Everything seems all right :smile: .
*CFME QE Bot*

**Pull request merged**